### PR TITLE
feat: add a new plugin weather_bit

### PR DESCRIPTION
**Pull request by LinkW77**
# Pull Request

## PR Description

Add a new plugin weather_bit.

## Linked Issue

feat: Add plugin-weather_bit to provide current, historical, and forecast weather data #260

## Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Performance enhancement
- [ ] Code refactor
- [ ] Documentation update
- [ ] Other, please describe:

## Checklist

Before submitting this PR, please make sure:

- [x] I have read the [CONTRIBUTING.md](../CONTRIBUTING.md) guidelines.
- [x] I have tested my changes locally to ensure they are effective.
- [x] I have updated the necessary documentation (if applicable).

### CODE REVIEW

File: plugin/bundles/weather_bit/plugins/get_current_weather/plugin.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+from config import CONFIG
+
+
+class GetCurrentWeather(PluginHandler):
+    async def execute(self, credentials: BundleCredentials, plugin_input: PluginInput) -> PluginOutput:
+        lat: float = plugin_input.input_params.get("lat")
+        lon: float = plugin_input.input_params.get("lon")
+        units: str = plugin_input.input_params.get("units", "")
+        weather_bit_api_key: str = credentials.credentials.get("WEATHER_BIT_API_KEY")
+        unit_tmp = {"M": "Celsius", "S": "Scientific", "I": "Fahrenheit"}
```
Comment (DynamesC) [inline plugin/bundles/weather_bit/plugins/get_current_weather/plugin.py:16]: "M" should be representing "Metric", as described in API documentation

File: plugin/bundles/weather_bit/plugins/get_current_weather/plugin.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+                    data = await response.json()
+                    content = data.get("data", [])[0]
+                    result = {
+                        "city_name": content["city_name"],
+                        "country_code": content["country_code"],
+                        "unit": unit,
+                        "temperature": content["temp"],
+                        "feels_like_temperature": content["app_temp"],
+                        "weather_condition": content["weather"]["description"],
+                        "aqi": content["aqi"],
```
Comment (DynamesC) [inline plugin/bundles/weather_bit/plugins/get_current_weather/plugin.py:38]: For extracting data from a dict, using `.get` with a default value is always safer than using square brackets

File: plugin/bundles/weather_bit/plugins/get_historical_weather/plugin.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        start_date: str = plugin_input.input_params.get("start_date")
+        end_date: str = plugin_input.input_params.get("end_date")
+        lat: float = plugin_input.input_params.get("lat")
+        lon: float = plugin_input.input_params.get("lon")
+        units: str = plugin_input.input_params.get("units")
+        tz: str = plugin_input.input_params.get("tz")
+        weather_bit_api_key: str = credentials.credentials.get("WEATHER_BIT_API_KEY")
+        unit_tmp = {"M": "Celsius", "S": "Scientific", "I": "Fahrenheit"}
+
+        url = f"https://api.weatherbit.io/v2.0/history/{interval}?lat={lat}&lon={lon}&start_date={start_date}&end_date={end_date}&key={weather_bit_api_key}"
```
Comment (DynamesC) [inline plugin/bundles/weather_bit/plugins/get_historical_weather/plugin.py:22]: Check start date and end date format

Comment (DynamesC) [reply plugin/bundles/weather_bit/plugins/get_historical_weather/plugin.py:22]: And interval value

File: plugin/bundles/weather_bit/plugins/get_current_weather/plugin.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        weather_bit_api_key: str = credentials.credentials.get("WEATHER_BIT_API_KEY")
+        unit_tmp = {"M": "Celsius", "S": "Scientific", "I": "Fahrenheit"}
+
+        url = f"https://api.weatherbit.io/v2.0/current?lat={lat}&lon={lon}&key={weather_bit_api_key}"
+
+        if units:
+            url += f"&units={units}"
+            unit = unit_tmp.get(units, "")
+        else:
+            unit = "Celsius"
```
Comment (DynamesC) [inline plugin/bundles/weather_bit/plugins/get_current_weather/plugin.py:24]: Default unit is 'Metric'

File: plugin/bundles/weather_bit/resources/i18n/en.yml [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+get_current_weather_input_lon_name: "Longitude"
+get_current_weather_input_lon_description: "The longitude of the location."
+get_current_weather_input_units_name: "Units"
+get_current_weather_input_units_description: "The units of the temperature. M - [DEFAULT](Celsius), S - (Kelvin), I - (Fahrenheit)"
+get_current_weather_output_city_name_name: "City Name"
+get_current_weather_output_city_name_description: "The name of the city."
+get_current_weather_output_country_code_name: "Country Code"
+get_current_weather_output_country_code_description: "The country code of the city."
+get_current_weather_output_unit_name: "Unit"
+get_current_weather_output_unit_description: "The units of the temperature. M - [DEFAULT](Celsius), S - (Kelvin), I - (Fahrenheit)"
```
Comment (DynamesC) [inline plugin/bundles/weather_bit/resources/i18n/en.yml:15]: Should be more specific and tell the model to only input a capital character, selecting from M, S, I

**Review by DynamesC — CHANGES_REQUESTED** [review]

**Review by DynamesC — APPROVED** [review]
LGTM

**Review by taskingaijc — APPROVED** [review]
LGTM

**Pull request merged**